### PR TITLE
run plugin hooks using powershell in windows

### PR DESCRIPTION
**Pull request by bacongobbler**
addresses #396, but doesn't close it until draft-pack-repo supports this as well (working on a PR, should be ready by tomorrow)

### CODE REVIEW

**Comment by bacongobbler** [comment]
currently untested, but PR is available here: https://github.com/Azure/draft-pack-repo/pull/14

**Comment by bacongobbler** [comment]
conflicts with #657 since we'll need Azure/draft-pack-repo#14 to support the $DRAFT_PLUGIN_VERSION envvar as well.

**Comment by bacongobbler** [comment]
OK, here's my test results from testing this PR on windows/linux using `draft init; rm -rf $HOME/.draft/plugins/draft-pack-repo; draft plugin install https://github.com/bacongobbler/draft-pack-repo --version powershell-support`

The default case won't work until Azure/draft-pack-repo#14 of course :)

Linux:

```
><> draft init
Creating /home/bacongobbler/.draft 
Creating /home/bacongobbler/.draft/plugins 
Creating /home/bacongobbler/.draft/packs 
Creating /home/bacongobbler/.draft/logs 
Creating /home/bacongobbler/.draft/config.toml 
Installing default plugins...
Error: error unmarshaling JSON: json: cannot unmarshal string into Go struct field Metadata.hooks of type plugin.Hooks
><> rm -rf ~/.draft/plugins/draft-pack-repo
><> draft plugin install https://github.com/bacongobbler/draft-pack-repo --version powershell-support
Downloading https://azuredraft.blob.core.windows.net/draft/pack-repo-canary-linux-amd64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3008k  100 3008k    0     0  1665k      0  0:00:01  0:00:01 --:--:-- 1665k
Preparing to install into /home/bacongobbler/.draft/plugins/draft-pack-repo
draft-pack-repo installed into /home/bacongobbler/.draft/plugins/draft-pack-repo/draft-pack-repo
Installed plugin: pack-repo
```

Just about to test Windows in a minute, lemme reboot

**Comment by bacongobbler** [comment]
Windows output:

```
Downloading https://github.com/Azure/draft-pack-repo/releases/download/canary/pack-repo-canary-windows-amd64.zip
Exception calling "DownloadFile" with "2" argument(s): "The request was aborted: Could not create SSL/TLS secure 
channel."
At C:\Users\mafishe\.draft\plugins\draft-pack-repo\scripts\get.ps1:13 char:1
+ (new-object System.Net.WebClient).DownloadFile($url, $file)
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException
 
Preparing to install into C:\Users\mafishe\.draft\plugins\draft-pack-repo\bin
Expand-Archive : The path 'C:\Users\mafishe\AppData\Local\Temp\pack-repo-canary-windows-amd64.zip' either does not 
exist or is not a valid file system path.
At C:\Users\mafishe\.draft\plugins\draft-pack-repo\scripts\get.ps1:19 char:1
+ Expand-Archive -Path "$file" -DestinationPath "$tempDir" -Force
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (C:\Users\mafish...ndows-amd64.zip:String) [Expand-Archive], InvalidOpe 
   rationException
    + FullyQualifiedErrorId : ArchiveCmdletPathNotFound,Expand-Archive
 
Move-Item : Cannot find path 'C:\Users\mafishe\AppData\Local\Temp\pack-repo\windows-amd64\pack-repo.exe' because it 
does not exist.
At C:\Users\mafishe\.draft\plugins\draft-pack-repo\scripts\get.ps1:20 char:1
+ Move-Item -Path "$tempDir\windows-amd64\$name.exe" -Destination "$ins ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\mafish...4\pack-repo.exe:String) [Move-Item], ItemNotFoundExce 
   ption
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.MoveItemCommand
 
pack-repo installed into C:\Users\mafishe\.draft\plugins\draft-pack-repo\bin\pack-repo.exe
Installed plugin: pack-repo

```

This should be fixed with https://github.com/Azure/draft-pack-repo/pull/14

**Comment by bacongobbler** [comment]
removing the WIP label as this is ready for review.

File: cmd/draft/testdata/plugins/echo/plugin.yaml [hunk 1]
```diff
@@ -4,4 +4,7 @@ description: "This echos stuff"
 command: "echo hello"
 version: "1.0.0"
 hooks:
+  bash:
     install: "echo installing"
+  powershell:
```
Comment (michelleN) [inline cmd/draft/testdata/plugins/echo/plugin.yaml]: 👍

**Comment by bacongobbler** [comment]
If anyone has time, I need to get the following PRs through before I can merge this:

- https://github.com/Azure/draft-pack-repo/pull/13
- https://github.com/Azure/draft-pack-repo/pull/14
- https://github.com/Azure/draft-pack-repo/pull/15

Thanks!

**Comment by bacongobbler** [comment]
k i'll rebase this and then merge both PRs at once

**Comment by bacongobbler** [comment]
for anyone doing dev work, please make sure to rebase off master with this change before running `draft init`. The official releases should still wor, but the canaries fetch the canary release of Draft and this is a backwards-incompatible change.

**Pull request merged**